### PR TITLE
Renaming iterator overheads

### PR DESCRIPTION
**Pull request by omatthew98**
## Why are these changes needed?
This renames the existing overheads to make it more clear what they mean. Changes are:
* `Total time user code is blocked` -> `Total time user thread is blocked by iter_batches`
* `Total time in user code` -> `Total execution time for user thread`

## Checks
- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/_internal/stats.py [hunk 1]
```diff
@@ -1173,11 +1173,13 @@ def to_string(self) -> str:
         ):
             out += "\nDataset iterator time breakdown:\n"
             if self.block_time.get():
-                out += "* Total time user code is blocked: {}\n".format(
-                    fmt(self.block_time.get())
+                out += (
+                    "* Total time user thread is blocked by iter_batches: {}\n".format(
+                        fmt(self.block_time.get())
```
Comment (scottjlee) [inline python/ray/data/_internal/stats.py:1178]: let's also rename the `self.block_time` and `self.user_time` to a slightly more descriptive name as well

File: python/ray/data/_internal/stats.py [hunk 2]
```diff
@@ -1173,11 +1173,13 @@ def to_string(self) -> str:
...
+                    )
                 )
             if self.user_time.get():
-                out += "* Total time in user code: {}\n".format(
+                out += "* Total execution time for user thread: {}\n".format(
```
Comment (scottjlee) [inline python/ray/data/_internal/stats.py:1182]: This is a personal opinion, but i feel that "user code" is more intuitive than "user thread" for the purpose of debugging logs. We should see what others think about the exact text we use

**Review by scottjlee — APPROVED** [review]

File: python/ray/data/_internal/stats.py (hunk 2 above)
Comment (scottjlee) [reply python/ray/data/_internal/stats.py:1182]: @stephanie-wang @c21 thoughts on the text description? "user code" vs "user thread" or any other alternative?

**Comment by omatthew98** [comment]
Closing this PR and incorporating changes from here into https://github.com/ray-project/ray/pull/42810.

**Pull request closed without merging**